x/authz/codec.go

```diff
@@ -1,11 +1,35 @@
 package authz
 
 import (
+	"github.com/cosmos/cosmos-sdk/codec"
 	types "github.com/cosmos/cosmos-sdk/codec/types"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	"github.com/cosmos/cosmos-sdk/types/msgservice"
 )
 
+// RegisterLegacyAminoCodec registers the necessary x/authz interfaces and concrete types
+// on the provided LegacyAmino codec. These types are used for Amino JSON serialization.
+func RegisterLegacyAminoCodec(cdc *codec.LegacyAmino) {
+	cdc.RegisterConcrete(&MsgGrant{}, "cosmos-sdk/MsgGrant", nil)
+	cdc.RegisterConcrete(&MsgRevoke{}, "cosmos-sdk/MsgRevoke", nil)
+	cdc.RegisterConcrete(&MsgExec{}, "cosmos-sdk/MsgExec", nil)
+
+	cdc.RegisterInterface((*Authorization)(nil), nil)
+	cdc.RegisterConcrete(&GenericAuthorization{}, "cosmos-sdk/GenericAuthorization", nil)
```

Comment:
We still have a problem here. GenericAuthorization allows any msg TypeURL, and then you can MsgExec with any arbitrary Msg. We would need to register `authz.RegisterConcrete` for every module.

Reply:
I don't know how to solve this problem right now. We can try to think of something before merging.

Reply:
OMG, and we need to use the right type name...